rero_ils/modules/documents/listener.py

```diff
@@ -116,22 +116,20 @@ def enrich_document_data(sender, json=None, record=None, index=None,
                     if n['type'] in ItemNoteTypes.PUBLIC
                 ]
                 if public_notes_content:
-                    item_record['notes'] = public_notes_content
+                    item_data['notes'] = public_notes_content
 
                 # related collection
                 #   index the collection title and description
-                item_obj = Item.get_record_by_pid(item['pid'])
                 for collection in item_obj.in_collection():
                     coll_data = {
                         'title': collection.get('title'),
                         'description': collection.get('description')
                     }
                     coll_data = {k: v for k, v in coll_data.items() if v}
-                    item_record.setdefault('collections', []).append(coll_data)
-
-                data.setdefault('items', []).append(item_record)
-            data['available'] = Holding.isAvailable(es_items)
-            holdings.append(data)
+                    item_data.setdefault('collections', []).append(coll_data)
+                hold_data.setdefault('items', []).append(item_data)
+            hold_data['available'] = holding_obj.available
```

Comment:
We have already all the items, it should be possible to calculate the availability without the Holding object directly in the items loop. Maybe we can get some speed improvements here.

Reply:
You're right. But `Holding.available` method has its own logic (really simple but centralized in one place).
I tried to pass items array to the `available` method, but we can ensure every time that this array contains all items for this holding ; in this case, this method will return a non-consistent value.